torch/csrc/distributed/c10d/socket.cpp

```diff
@@ -650,20 +653,26 @@ bool SocketConnectOp::tryConnect(int family) {
         return true;
       }
 
-      if (cr == ConnectResult::TimeOut) {
-        auto msg = fmt::format(
-            "The client socket has timed out after {} while trying to connect to ({}, {}).",
-            opts_->connect_timeout(),
-            host_,
-            port_);
+      if (cr == ConnectResult::Retry) {
```

Comment:
nit: can we merge this into L661?

Reply:
This is inside the `do`/`while` loop, so at this point we do not know whether another IP address in our stack will succeed before we retry from start. This keeps record whether we should retry it after we exhaust all our options.